packages/@aws-cdk/cloud-assembly-schema/lib/assembly-manifest.ts

```diff
@@ -0,0 +1,340 @@
+interface BaseAssetMetadataEntry {
+    /**
+     * Requested packaging style
```

Comment:
Let's expand an explain what a 'packaging style' is.

Let's go through all the documentation in this new package and make sure it's easy-to-understand.

Reply:
This might take a while...do we want it to block us? im not sure...most important thing in the PR are the interfaces themselves, we can improve the docs and everything else later on. (famous last words)

Reply:
Why not now? This is a core piece of our system. Holding it to a high standard would pay off, no?

Of course, unless there's a real rush to get this out.

Reply:
added the doc